samcli/lib/cli_validation/image_repository_validation.py

```diff
@@ -103,6 +82,17 @@ def wrapped(*args, **kwargs):
                         option_name="--image-repositories", ctx=ctx, message=image_repos_error_msg
                     ),
                 ),
+                Validator(
+                    validation_function=lambda: not guided
+                    and not image_repositories
+                    and not _is_all_image_funcs_provided(template_file, {}, parameters_overrides)
```

Comment:
I think we still need to update `_is_all_image_funcs_provided` function logic. The current logic of this function is to check if all Image type functions defined in the template are exist in the `image_repositories` list. It does not exclude the functions that has an ECR url value.

Reply:
To my understanding, it actually does (?)

To acquire the list of Lambda functions from the template, it relies on [SamFunctionProvider](https://github.com/aws/aws-sam-cli/blob/98f8f3b738c5a98e159e04a899f0784a567daa43/samcli/lib/providers/sam_function_provider.py#L179) which ignores IMAGE Lambdas with ECR registries

Reply:
my bad, I did not check the Function Provider.